src/Database/LSMTree/Internal/WriteBuffer.hs

```diff
@@ -36,21 +38,32 @@ import           Database.LSMTree.Common (Range (..))
 import           Database.LSMTree.Internal.Entry
 import qualified Database.LSMTree.Internal.Monoidal as Monoidal
 import qualified Database.LSMTree.Internal.Normal as Normal
+import           Database.LSMTree.Internal.Serialise
 
 {-------------------------------------------------------------------------------
   Writebuffer type
 -------------------------------------------------------------------------------}
 
-newtype WriteBuffer k v blobref = WB (Map k (Entry v blobref))
+newtype WriteBuffer k v blob =
```

Comment:
I'm now wondering whether we need these type parameters at this level or if we can monomorphise everything to `Serialised*` types. i.e., similar to how `CompactIndex`, `Bloom`, `RunAcc`, `PageAcc`, etc. are monomorphic over `SerialisedKey`s. Instead, we only add the type parameters at the top level at the `TableHandle`

It probably simplifies the code here a little bit, and moves the responsibility for serialisation up to the `TableHandle` entrypoint

Reply:
We probably don't need them, as it's an internal API. I'm still tempted to leave it here for now until we're writing the code using it. It might still bring some type safety and the simplification would only really be in the type signatures, as the serialisation still needs to be done somewhere else.

I do get the consistency argument, though, and will at least leave a comment.